### PR TITLE
adding initial mqtt support

### PR DESCRIPTION
**Pull request by yGuy**
This adds simple MQTT publishing support. You can configure an MQTT broker to which all properties will be published to.

### CODE REVIEW

File: src/EnergyExporter/Options/ExportOptions.cs [hunk 1]
```diff
@@ -26,4 +29,20 @@ public class InfluxDbOptions
...
+    public class MqttOptions
+    {
+        [Required]
+        public string TcpServer { get; init; } = null!;
+
+        public int Port { get; init; } = 1883;
+
+        public string ClientId { get; init; } = "EnergyExplorer";
+        
+        public string Topic { get; init; } = "EnergyExplorer";
```
Comment (MarcusWichelmann) [inline src/EnergyExporter/Options/ExportOptions.cs:42]: You probably meant `EnergyExporter`?

File: src/EnergyExporter/appsettings.sample.json [hunk 2]
```diff
@@ -59,12 +59,22 @@
...
       "Bucket": "Solar",
       "Organisation": "MyOrganization",
       "Token": "Token"
+    },
+    "Mqtt": {
+      "TcpServer": "mqtt.domain.tld",
+      // the following is all optional inside this section. Defaults are shown, here
+      "Port": 1883,
+      "ClientId": "EnergyExplorer",
+      "Topic": "EnergyExplorer",
```
Comment (MarcusWichelmann) [inline src/EnergyExporter/appsettings.sample.json:74]: Same here. `EnergyExporter` maybe?

File: src/EnergyExporter/MQTT/MqttExporter.cs [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+using EnergyExporter.InfluxDb;
+using EnergyExporter.Options;
+using EnergyExporter.Services;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using MQTTnet;
+using MQTTnet.Client;
+using MQTTnet.Protocol;
+
+namespace EnergyExporter.MQTT;
```
Comment (MarcusWichelmann) [inline src/EnergyExporter/MQTT/MqttExporter.cs:17]: Per convention, I think `EnergyExporter.Mqtt` would be more correct (also rename directory please).

**Review by MarcusWichelmann — CHANGES_REQUESTED** [review]
Wow, thank you very much for your contribution. Nice to also have MQTT support.
Also great to see, that this project is useful for others, too. 😄 

Only found a few minor things, would be great if you could fix them, so we can get this merged.

Looks like it may make sense, to generalize some of the stuff you adopted from the InfluxDbExporter (the attribute stuff etc.), but I think that's good enough for now. I may look into that sometime in the future.

File: src/EnergyExporter/Options/ExportOptions.cs (hunk 1 above)
Comment (yGuy) [reply src/EnergyExporter/Options/ExportOptions.cs]: Whoops :-)   - sorry, that wasn't on purpose. Corrected.

**Comment by MarcusWichelmann** [comment]
Thank you. LGTM!

**Pull request merged**